### PR TITLE
str_split() length parameter is always positive

### PR DESCRIPTION
**Pull request by staabm**
https://3v4l.org/TNDjR

inspired by https://github.com/vimeo/psalm/pull/6324

### CODE REVIEW

File: resources/functionMap.php [hunk 1]
```diff
@@ -11696,7 +11696,7 @@
 'str_replace' => ['string|array', 'search'=>'string|array', 'replace'=>'string|array', 'subject'=>'string|array', '&w_replace_count='=>'int'],
 'str_rot13' => ['string', 'str'=>'string'],
 'str_shuffle' => ['string', 'str'=>'string'],
-'str_split' => ['array<int,string>|false', 'str'=>'string', 'split_length='=>'int'],
+'str_split' => ['array<int,string>|false', 'str'=>'string', 'split_length='=>'positive-int'],
```
Comment (staabm) [inline resources/functionMap.php:11699]: `str_split` always returns a non-empty array.. I tried adding this here, but it seems the `&nonEmpty` signature doesn't work from this file

Comment (staabm) [reply resources/functionMap.php:11699]: since maybe controversial, put it into a separate PR:
https://github.com/phpstan/phpstan-src/pull/632

Comment (ondrejmirtes) [reply resources/functionMap.php:11699]: The correct is `'non-empty-array<int,string>|false'`

**Comment by ondrejmirtes** [comment]
Thank you.

File: resources/functionMap.php (hunk 1 above)
Comment (staabm) [reply resources/functionMap.php:11699]: ahh makes sense.. https://github.com/phpstan/phpstan-src/pull/633

**Pull request merged**